c10/core/impl/DeviceGuardImplInterface.h

```diff
@@ -212,6 +220,15 @@ struct C10_API DeviceGuardImplInterface {
     TORCH_CHECK(false, "Backend doesn't support synchronizing events.");
   }
 
+  /**
+   * Wait (by blocking the calling thread) until all the work previously
+   * enqueued on the all streams has completed running on the device.
+   */
+  virtual void syncStreamsOnDevice(const DeviceIndex /*device_index*/) const {
```

Comment:
nit: what about synchronizeDevice() to mimic other functions here?

Reply:
`synchronizeDevice` sounds better. They are so consistent with `synchronizeStream` and `synchronizeEvent`.